src/packaging_tool_versions.rs

```diff
@@ -2,6 +2,8 @@ use serde::{Deserialize, Serialize};
 
 /// The versions of various packaging tools used during the build.
 /// These are always installed, and are independent of the chosen package manager.
+/// Strings are unused instead of a semver version, since these packages don't use
+/// semver, and we never introspect the version parts anyway.
```

Comment:
FWIW, it you needed to work with the versions eventually, someone implemented the python packaging version spec: https://crates.io/crates/pep440

Reply:
@merwok Yeah I'd spotted that crate while looking for another use-case - great that it exists! (Albeit sadly it only parses single versions and not ranges, so it can't be used for things like parsing python version ranges from the PEP440 style versions in a Poetry lockfile.)